WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift

```diff
@@ -306,7 +306,8 @@ extension DashboardViewModel {
     private func isBlazeBannerVisible() async -> Bool {
         async let isSiteEligible = blazeEligibilityChecker.isSiteEligible()
         async let storeHasPublishedProducts = (try? checkIfStoreHasProducts(siteID: siteID, status: .published)) ?? false
-        guard (await isSiteEligible, await storeHasPublishedProducts) == (true, true) else {
+        async let storeHasAnyOrders = (try? checkIfStoreHasOrders(siteID: siteID)) ?? false
+        guard await(isSiteEligible, storeHasPublishedProducts, storeHasAnyOrders) == (true, true, false) else {
```

Comment:
nit: Would it make sense to introduce a separate file to reuse this `isBlazeBannerVisible` logic in dashboard and product list screens?

Reply:
The logic is specific to each view model, it just happens that both screens show the same banner, so I'd prefer keeping the checks separate. Plus, `checkIfStoreHasProducts` is needed in `DashboardViewModel` for checking product onboarding eligibility, so we can't remove it. I hope we'll remove the banners soon and then remove the code for these checks.

I'll proceed to merge this PR for now, thanks.